integrations/docs/Integration-API.md

```diff
@@ -0,0 +1,285 @@
+
+Integrations are a stateful bundle which will be stored inside a system `.integration` index which will reflect the integration's status during the different phases of its lifecycle.
+
+---
+## Integration UX Loading Lifecycle API 
+
+### Flow Diagram
+![Screenshot 2023-03-01 at 7 00 50 PM](https://user-images.githubusercontent.com/48943349/222320100-cac40749-9e5a-4e90-8ff2-386958adc06d.png)
+
+### Load Integrations Repositoty
+As part of the Integration Ux workflow, once the Integration plugin is loaded it should load all the available integrations that are bundled in the integration repo.
+
+![Screenshot 2023-03-01 at 10 01 45 AM](https://user-images.githubusercontent.com/48943349/222223963-9c740f33-e156-4541-88cf-67d70131410f.png)
+
+
+The API needed from the backend should be as follows:
+Query:
+```text
+GET _integration/repository?filter=type:Logs&category:web,html
+```
+**Response**:
+```jsoon
+{
+  "name": "nginx",
+  "version": {
+    "integ": "0.1.0",
+    "schema": "1.0.0",
+    "resource": "^1.23.0"
+  },
+  "description": "Nginx HTTP server collector",
+  "Information":"file:///.../schema/logs/info.html",
+  "identification": "instrumentationScope.attributes.identification",
+  "categories": [
+    "web","http"
+  ],
+  "collection":[
+    {
+      "logs": [{
+        "info": "access logs",
+        "input_type":"logfile",
+        "dataset":"nginx.access",
+        "labels" :["nginx","access"],
+        "schema": "file:///.../schema/logs/access.json"
+      },
+        {
+          "info": "error logs",
+          "input_type":"logfile",
+          "labels" :["nginx","error"],
+          "dataset":"nginx.error",
+        "schema": "file:///.../schema/logs/error.json"
+        }]
+    },
+    {
+      "metrics": [{
+        "info": "status metrics",
+        "input_type":"metrics",
+        "dataset":"nginx.status",
+        "labels" :["nginx","status"],
+        "schema": "file:///.../schema/logs/status.json"
+      }]
+    }
+  ],
+  "repo": {
+    "github": "https://github.com/opensearch-project/observability/tree/main/integrarions/nginx"
+  }
+}
+```
+The integration object schema is supported by both B/E & F/E for display & query to the correct fields
+[Integration config schema](https://github.com/opensearch-project/observability/blob/9a22f061f568443651fe38d96c864901eed12340/schema/system/integration.schema)
+
+---
+The backend responsibilities :
+- scan the Integration folder (on-load)
+    - In the future this can also be loaded from a remote publish location
+
+- load into cache each integration config file
+- allow filtering on the registry API
+
+The frontend responsibilities :
+- enable shared info (html, images) resources in a CDN
+- allow filtering for the integrations
+
+![Screenshot 2023-03-01 at 10 13 41 AM](https://user-images.githubusercontent.com/48943349/222226930-1d9a684d-7f19-4aaf-b601-32bf7ce08920.png)
+
+In addition the following API is also supported
+```
+GET _integration/repository/$name
```

Comment:
what if integrations (observability backend) doesn't exist ?
Will we need to split this off into its own plugin that HAS to exist? Do we need to bring this into core?

Reply:
all catalog schema are loaded during integration startup so once theoretically all schema's exist once Integration is loaded - 
If an integration will attempt to use schema which wasnt loaded for any reason it will result the appropriate error reflecting this status

see  https://github.com/opensearch-project/observability/blob/64eb7fe6889689dbbf6e06e687f6378efe7a3085/integrations/docs/Integration-plugin-tasks.md#catalog-registration

Reply:
Integration would be marked as a [mandatory plugin](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/plugins/PluginsService.java#L201)